ui-admin/src/util/tableUtils.tsx

```diff
@@ -1,23 +1,101 @@
 import React, { HTMLProps, useState } from 'react'
-import { flexRender, Header, Table } from '@tanstack/react-table'
+import { Column, flexRender, Header, Table } from '@tanstack/react-table'
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
 import { faCaretDown, faCaretUp, faColumns } from '@fortawesome/free-solid-svg-icons'
 
+/**
+ * Returns a debounced input react component
+ * Adapted from https://tanstack.com/table/v8/docs/examples/react/filters
+ */
+function DebouncedInput({
+  value: initialValue,
+  onChange,
+  debounce = 200,
+  ...props
+}: {
+  value: string
+  onChange: (value: string) => void
+  debounce?: number
+} & Omit<React.InputHTMLAttributes<HTMLInputElement>, 'onChange'>) {
+  const [value, setValue] = React.useState(initialValue)
+
+  React.useEffect(() => {
+    setValue(initialValue)
+  }, [initialValue])
+
+  React.useEffect(() => {
+    const timeout = setTimeout(() => {
+      onChange(value)
+    }, debounce)
+
+    return () => clearTimeout(timeout)
+  }, [value])
+
+  return (
+    <input {...props} value={value} onChange={e => setValue(e.target.value)} />
+  )
+}
+
+/**
+ * returns a Filter to handle text fields
+ * adapted from https://tanstack.com/table/v8/docs/examples/react/filters
+ * */
+function Filter<A>({
+  column
+}: {
+  column: Column<A>
+}) {
+  const columnFilterValue = column.getFilterValue()
+
+  return <div>
+    <DebouncedInput
+      type="text"
+      value={(columnFilterValue ?? '') as string}
+      onChange={value => column.setFilterValue(value)}
+      placeholder={`Search...`}
+      list={`${column.id} list`}
```

Comment:
What's responsible for rendering the datalist element referenced here?

Reply:
Ah this is a leftover from the react-table example code. They had a bunch of handling for array column types, which we don't need. Just forgot to remove this line. Done in 00d83beaf9d7a364313eced7fae712335cb0c5df